parser/prism/Translator.cc

```diff
@@ -1356,7 +1356,31 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
 
             auto returnValues = translateArguments(returnNode->arguments);
 
-            return make_unique<parser::Return>(location, move(returnValues));
+            if (returnValues.empty()) {
+                auto expr = MK::Return(location, MK::EmptyTree());
+                return make_node_with_expr<parser::Return>(move(expr), location, move(returnValues));
+            }
+
+            if (!hasExpr(returnValues)) {
```

Comment:
Do we need `!directlyDesugar` here as with the other cases?

```suggestion
            if (!directlyDesugar || !hasExpr(returnValues)) {
```